### PR TITLE
Update ICU breaking change with practical examples

### LINKED ISSUE #19866: Add practical examples to NLS -> ICU breaking change
For example:

- https://github.com/dotnet/runtime/issues/38308 (currency symbol)
- https://github.com/dotnet/runtime/issues/37229

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 5f68af3a-3848-48ef-c24b-cf33b880ca1b
* Version Independent ID: 580c061d-20e5-248c-a24f-7a48a684c7e3
* Content: [Breaking changes, version 3.1 to 5.0 - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/compatibility/3.1-5.0#globalization-apis-use-icu-libraries-on-windows)
* Content Source: [docs/core/compatibility/3.1-5.0.md](https://github.com/dotnet/docs/blob/master/docs/core/compatibility/3.1-5.0.md)
* Product: **dotnet-core**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #21249.
Fixes #19866.

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/3.1-5.0?branch=pr-en-us-21345#globalization-apis-use-icu-libraries-on-windows).

### CODE REVIEW

**Review by safern — APPROVED** [review]
LGTM. Thanks! 

cc: @tarekgh

File: includes/core-changes/globalization/5.0/icu-globalization-api.md [hunk 1]
```diff
@@ -4,14 +4,51 @@
 
 #### Change description
 
-Previously, .NET libraries used [National Language Support (NLS)](/windows/win32/intl/national-language-support) APIs for globalization functionality. For example, NLS functions were used to get culture data, such as date and time format patterns, compare strings, and perform string casing in the appropriate culture.
+In previous .NET versions, .NET libraries use [National Language Support (NLS)](/windows/win32/intl/national-language-support) APIs for globalization functionality on Windows. For example, NLS functions were used to compare strings, get culture information, and perform string casing in the appropriate culture.
```
Comment (tarekgh) [inline includes/core-changes/globalization/5.0/icu-globalization-api.md:7]: This is not true for .NET Framework 1.0, 2.0, 3.0 and 3.5. Maybe we can say Full Framework versions 4.0 and above and .NET Core.

**Review by tarekgh — APPROVED** [review]
I added a small comment. other than that LGTM.

File: includes/core-changes/globalization/5.0/icu-globalization-api.md (hunk 1 above)
Comment (gewarren) [inline includes/core-changes/globalization/5.0/icu-globalization-api.md:7]:
```suggestion
In .NET Core 1.0 - 3.1 and .NET Framework 4 and later, .NET libraries use [National Language Support (NLS)](/windows/win32/intl/national-language-support) APIs for globalization functionality on Windows. For example, NLS functions were used to compare strings, get culture information, and perform string casing in the appropriate culture.
```

**Comment by gewarren** [comment]
Thanks for the reviews!

**Pull request merged**